src/main/java/org/jahia/modules/sitemap/filter/SitemapCacheFilter.java

```diff
@@ -152,7 +149,20 @@ public boolean needsCaching(Resource resource) {
     }
 
     public boolean isValidCache(JCRNodeWrapper cacheNode) {
-        return (cacheNode != null) && !CacheUtils.isExpired(cacheNode, EXPIRATION);
+        long expiration = 4 * 60 * 60 * 1000L; // Default to 4 hours
+        if (cacheNode == null) return false;
+        try {
+            JCRNodeWrapper siteNode = cacheNode.getResolveSite();
+            if (siteNode != null) {
+                JCRPropertyWrapper cacheDurationProperty = siteNode.getProperty(SITEMAP_CACHE_DURATION);
+                String propertyValue = ConversionUtils.getValueFromJCRProperty(cacheDurationProperty);
+                expiration = ConversionUtils.toMilliSecondsLong(propertyValue, expiration);
+            }
+        } catch (RepositoryException e) {
+            logger.error("Unable to retrieve node information.");
+            return false;
+        }
+        return !CacheUtils.isExpired(cacheNode, expiration);
```

Comment:
add validation that expiration value is valid (i.e. >= 0) perhaps on isExpired() )

Reply:
I will add a check to skip and state that is it expired when expiration is < 0